flask_resty/testing.py

```diff
@@ -27,28 +50,39 @@ def get_meta(response):
     return get_body(response)['meta']
 
 
-def assert_value(actual, expected):
+def assert_similar(actual, expected):
     if isinstance(expected, Mapping):
         assert isinstance(actual, Mapping)
         # Unlike all the others, this checks that the actual items are a
         # superset of the expected items, rather than that they match.
         for key, value in expected.items():
             if value is not UNDEFINED:
                 assert key in actual
-                assert_value(actual[key], value)
+                assert_similar(actual[key], value)
             else:
                 assert key not in actual
     elif isinstance(expected, basestring):
-        assert actual == expected
+        assert expected == actual
```

Comment:
similar thread here: http://osdir.com/ml/python-testing-general/2012-01/msg00025.html

the way they solved it was to invert the order of comparison, like I did here